common-libs/common/src/services/DidDocumentService/ElemAnchoredDidDocument.ts

```diff
@@ -0,0 +1,26 @@
+import KeysService from '../KeysService'
+import { DidResolver } from '../../shared/DidResolver'
+
+export default class ElemDidDocument {
+  private readonly _keysService: KeysService
+  private readonly _signingKey: string
+  private readonly _didResolver: DidResolver
+
+  constructor(keysService: KeysService, didResolver: DidResolver) {
+    this._signingKey = 'primary'
+    this._keysService = keysService
+    this._didResolver = didResolver
+  }
+
+  getMyDid(): string {
+    const { anchoredDid } = this._keysService.getMetadata()
```

Comment:
And `getMetadata` decrypts a seed on every call. Is it really a good idea to do that whenever we need to obtain DID?

Maybe it would be better to use KeyVault here instead of KeysService?

Reply:
Not sure, I'm new to this codebase, but as I found, there are a lot of invocations of decryptSeed method
Also, not sure how often it will be invoked, maybe even less than other methods that uses decryptSeed (methods in KeysService).

If you really know that it will affect performance in some places, please help me find solution, in other case, let's use this rule:   
“The real problem is that programmers have spent far too much time worrying about efficiency in the wrong places and at the wrong times; premature optimization is the root of all evil (or at least most of it) in programming.” - The Art of Computer Programming,  Donald Knuth

Reply:
@anton-iskryzhytskyi , I'm not talking about premature optimization here, but we have a higher-level `KeyVault` already, and it seems to be a better place for that kind of functionality than `KeysService`.

Additionally, do we know how this should interact with other kinds of `KeyVault`? If we need this to work with W3 wallet properly, we might need to use a generic `KeyVault` anyway.

Reply:
Still no answer for:

> Additionally, do we know how this should interact with other kinds of KeyVault? If we need this to work with W3 wallet properly, we might need to use a generic KeyVault anyway.